contrib/pg_tde/expected/create_database_1.out

```diff
@@ -0,0 +1,105 @@
+CREATE EXTENSION IF NOT EXISTS pg_tde;
+CREATE DATABASE template_db;
+SELECT current_database() AS regress_database
+\gset
+\c template_db
+CREATE EXTENSION pg_tde;
+SELECT pg_tde_add_database_key_provider_file('file-vault', '/tmp/template_provider.per');
+ pg_tde_add_database_key_provider_file 
+---------------------------------------
+                                     1
+(1 row)
+
+SELECT pg_tde_set_key_using_database_key_provider('test-db-key', 'file-vault');
+ pg_tde_set_key_using_database_key_provider 
+--------------------------------------------
+ 
+(1 row)
+
+CREATE TABLE test_enc (id serial PRIMARY KEY, x int) USING tde_heap;
+CREATE TABLE test_plain (id serial PRIMARY KEY, x int) USING heap;
+INSERT INTO test_enc (x) VALUES (10), (20);
+INSERT INTO test_plain (x) VALUES (30), (40);
+\c :regress_database
+CREATE DATABASE new_db TEMPLATE template_db;
+SELECT pg_tde_add_global_key_provider_file('global-file-vault','/tmp/template_provider_global.per');
+ pg_tde_add_global_key_provider_file 
+-------------------------------------
+                                  -5
+(1 row)
+
+SELECT pg_tde_set_default_key_using_global_key_provider('default-key', 'global-file-vault');
+ pg_tde_set_default_key_using_global_key_provider 
+--------------------------------------------------
+ 
+(1 row)
+
+CREATE DATABASE new_db TEMPLATE template_db;
+ERROR:  database "new_db" already exists
```

Comment:
Something seems to be wrong in this testcase - this database already exists.

The description also said it only works if we have a default key, but we didn't have one during the previous create db.

Reply:
This seems possibly like a bug unrelated to the PR. And since it only happens when when TDE is enabled for the whole test suite it will not directly be an issue now that that has been changed but I think we should take a look and try to figure out if I accidentally found a bug here.

Reply:
No, the bug was in my test script. Doh. The issue is that we cannot delete the current default principal key. These tests used to run before the default key provider tests.

Reply:
Fixed by adding a comment and commenting out the code.